project/src/Player/PlayerShip.gd

```diff
@@ -1,7 +1,39 @@
 extends KinematicBody2D
 
 
+signal damaged(amount)
+signal player_dead
+
+
+export var health := 100
+export(int, LAYERS_2D_PHYSICS) var projectile_mask := 0
+export var PopEffect: PackedScene
+
 var can_dock := false
+var _current_health := health
+var dockable: Node2D
 
 onready var shape := $CollisionShape
 onready var agent: GSTSteeringAgent = $StateMachine/Move.agent
+onready var effects_parent: Node2D = get_tree().get_nodes_in_group("Effects")[0]
+onready var camera := $Camera2D
+
+
+func _ready() -> void:
+	connect("damaged", self, "_on_self_damaged")
+
+
+func _on_self_damaged(amount: int) -> void:
+	_current_health -= amount
+	if _current_health <= 0:
+		var effect := PopEffect.instance()
+		effect.global_position = global_position
+		effects_parent.add_child(effect)
+		
+		emit_signal("player_dead")
+		
+		remove_child(camera)
```

Comment:
Better separate the camera from the player in the first place than reparent the node at runtime, or have another camera that becomes current when the player dies. In general, I'd prefer the hierarchy of scenes to not mutate at runtime, i.e. nodes to reparent or move too much, it makes the game harder to debug